bin/internal/update_dart_sdk.sh

```diff
@@ -60,25 +60,50 @@ if [ ! -f "$ENGINE_STAMP" ] || [ "$ENGINE_VERSION" != `cat "$ENGINE_STAMP"` ]; t
   }
 
   # `uname -m` may be running in Rosetta mode, instead query sysctl
-  if [ "$OS" = 'Darwin' ]; then
+  if [ -z "$ARCH" ] && [ "$OS" = 'Darwin' ]; then
     # Allow non-zero exit so we can do control flow
     set +e
-    # -n means only print value, not key
-    QUERY="sysctl -n hw.optional.arm64"
-    # Do not wrap $QUERY in double quotes, otherwise the args will be treated as
-    # part of the command
-    QUERY_RESULT=$($QUERY 2>/dev/null)
-    if [ $? -eq 1 ]; then
-      # If this command fails, we're certainly not on ARM
-      ARCH='x64'
-    elif [ "$QUERY_RESULT" = '0' ]; then
-      # If this returns 0, we are also not on ARM
-      ARCH='x64'
-    elif [ "$QUERY_RESULT" = '1' ]; then
-      ARCH='arm64'
-    else
-      >&2 echo "'$QUERY' returned unexpected output: '$QUERY_RESULT'"
-      exit 1
+    if command -v arch >/dev/null 2>&1; then
```

Comment:
I wonder if you even need to run arch? That just tells you if you're running in Rosetta, and you don't really care about that, you care what the capabilities of the machine are.

```bash
if [[ "$(sysctl -n hw.cpu64bit_capable 2> /dev/null)" == 1 || "$(uname -m)" =~ arm64 ]]; then
  # It's an ARM machine.
elif  [[ "$(uname -m)" =~ 86 ]]; then
  # It's x86 of some sort.
fi
```

Reply:
I'm not on a mac right now, but I thought `arch` reported your actual physical architecture, and had nothing to do with rosetta, per https://stackoverflow.com/questions/65259300/detect-apple-silicon-from-command-line.

Reply:
decided to not use arch at all